build-push-ecr/action.yml

```diff
@@ -35,25 +47,70 @@ runs:
       with:
         role-to-assume: ${{ inputs.role-to-assume }}
         aws-region: ${{ inputs.aws-region }}
-    - run: echo "tag=${{ inputs.docker-repo }}:git-$(git rev-parse --short HEAD)" >> $GITHUB_OUTPUT
-      id: docker
-      shell: bash
-    - run: >
-        docker build
-        ${{ inputs.docker-additional-args }}
-        --pull -t ${{ steps.docker.outputs.tag }} ${{ inputs.dockerfile-path }}
-      shell: bash
-    - run: >
-        aws ecr get-login-password --region ${{ inputs.aws-region }}
-        | docker login --username AWS --password-stdin ${{ inputs.docker-repo }}
+    - name: Calculate tags
+      id: additional-tags
       shell: bash
-    - run: docker push ${{ steps.docker.outputs.tag }}
+      run: |
+        {
+          echo 'tags<<EOF'
+          if ${{ contains(inputs.docker-additional-tags, '=') }}; then
+            # already in the right format for docker/metadata-action
+            echo "${{ inputs.docker-additional-tags }}"
+          else
+            # pass tags in as raw
+            docker_additional_tags=(${{ inputs.docker-additional-tags }})
+            for tag in ${docker_additional_tags[@]}; do
+              echo "type=raw,priority=900,value=${tag}"
+            done
+          fi
+          echo 'EOF'
+        } | tee -a "$GITHUB_OUTPUT"
+      if: ${{ inputs.docker-additional-tags != '' }}
+    - name: Calculate build args
+      id: build-args
       shell: bash
-    - run: >
-        docker_additional_tags=(${{ inputs.docker-additional-tags }});
-        for tag in ${docker_additional_tags[@]}; do
-          docker tag ${{ steps.docker.outputs.tag }} ${{ inputs.docker-repo }}:$tag
-          docker push ${{ inputs.docker-repo }}:$tag
-        done
+      run: |
+        {
+          echo 'build-args<<EOF'
+          if ${{ inputs.docker-additional-args != '' }}; then
+            docker_additional_args=(${{ inputs.docker-additional-args }})
+            EXPECTED_FLAG="the --build-arg flag"
+            EXPECTED_ARG="an argument like FOO=bar"
+            arg_expected="$EXPECTED_FLAG"
+            for arg in ${docker_additional_args[@]}; do
+              if [[ "$arg_expected" = "$EXPECTED_FLAG" && "$arg" = "--build-arg" ]]; then
+                arg_expected="$EXPECTED_ARG"
+              elif [[ "$arg_expected" = "$EXPECTED_ARG" && "$arg" =~ "=" ]]; then
+                echo "$arg"
+                arg_expected="$EXPECTED_FLAG"
+              else
+                echo "Expected $arg_expected, got '$arg'" >&2
+                exit 1
+              fi
+            done
+          fi
+          echo "${{ inputs.docker-build-args }}"
+          echo 'EOF'
+        } | tee -a "$GITHUB_OUTPUT"
+      if: ${{ inputs.docker-additional-args != '' || inputs.docker-build-args != '' }}
+    - uses: docker/metadata-action@v5
+      id: meta
+      with:
+        images: ${{ inputs.docker-repo }}
+        tags: |
+          type=sha,priority=1000,prefix=git-
```

Comment:
question: does this generate the SHA tag in the same way the old script did?

Reply:
As it stands, it fetches the git SHA from the GitHub Actions context, which should still be the same thing since we aren't doing anything wacky like checking out a different commit than the workflow is actually running on. If we want to handle weird edge cases like that just in case, we can pass [context: git](https://github.com/docker/metadata-action#context-input), but I don't think there's any point in that.